doc/source/contribute.rst

```diff
@@ -131,107 +253,162 @@ For example: ::
       """
       return str(number) == string
 
+.. _contributing.testing:
 
-Documentation
--------------
+Step 4: Test your code
+~~~~~~~~~~~~~~~~~~~~~~
 
-The documentation is written using reStructuredText and built to various formats using
-`Sphinx <http://sphinx-doc.org/>`_. See the `reStructuredText Primer <http://sphinx-doc.org/rest.html#rst-primer>`_
-for a first introduction of the syntax.
+Sometimes doctests are not enough and new features require to go a step further by writing unit tests.
 
-Installing Requirements
-~~~~~~~~~~~~~~~~~~~~~~~
+Our unit tests are written using the `pytest library <https://docs.pytest.org>`_
+and our tests modules are located in `/larray/tests/`.
+The pytest library is able to automatically detect and run unit tests
+as long as you respect some conventions:
 
-Basic requirements (to generate an .html version of the documentation) can be installed using: ::
+  - pytest will search for ``test_*.py`` or ``*_test.py files``.
+  - From those files, collect test items:
 
-  > conda install sphinx numpydoc
+    - ``test_`` prefixed test functions or methods outside of class.
+    - ``test_`` prefixed test functions or methods inside Test prefixed test classes
+      (without an __init__ method).
 
-To build the .pdf version, you need a LaTeX processor. We use `MiKTeX <http://miktex.org>`_.
+For more details, please read the section `Conventions for Python test discovery
+<https://docs.pytest.org/en/latest/goodpractices.html#test-discovery>`_
+from the pytest documentation.
 
-To build the .chm version, you need `HTML Help Workshop
-<http://www.microsoft.com/en-us/download/details.aspx?id=21138>`_.
+Here is an example of a unit test function using pytest: ::
 
+  from larray.core.axis import _to_key
 
-Generating the documentation
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+  def test_key_string_split():
+      assert _to_key('M,F') == ['M', 'F']
+      assert _to_key('M,') == ['M']
 
-Open a command prompt and go to the documentation directory: ::
+To run unit tests for a given test module: ::
 
-  > cd doc
+  > pytest larray/tests/test_array.py
 
-If you just want to check that there is no syntax error in the documentation and that it formats properly, it is
-usually enough to only generate the .html version, by using: ::
+We also use doctests for some tests. Doctests is specially-formatted code within the docstring of a function which
```

Comment:
the order of paragraphs in this section is strange: "Sometimes doctests are not enough..." and then "we also use doctests for some tests".